index.d.ts

```diff
@@ -0,0 +1,208 @@
+    Attributes?: Schema & SchemaBase;
+  }
+  export class Entity<Schema> {
+    constructor(options: EntityConstructor);
+
+    // Properties
+    table: Table;
+    readonly DocumentClient: DocumentClient;
+    autoExecute: boolean;
+    autoParse: boolean;
+    readonly partitionKey: string;
+    readonly sortKey: string;
+
+    // Methods
+    attribute(attribute: string): string;
+    parse(input, include?);
+    get(
+      item: Schema,
+      options?: {
+        consistent?: boolean;
+        capacity?: "none" | "total" | "indexes";
+        attributes?: [] | {};
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<GetOuput<Schema>>;
+    delete(
+      key: {},
+      options?: {
+        conditions?: [] | {};
+        capacity?: "none" | "total" | "indexes";
+        metrics?: "none" | "size";
+        returnValues?: "none" | "all_old";
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<DeleteOutput<Schema>>;
+    put(
+      item: Schema,
+      options?: {
+        conditions?: [] | {};
+        capacity?: "none" | "total" | "indexes"; // Cannot use DynamoDB type as it's upper-case
+        metrics?: "none" | "size";
+        returnValues?: "none" | "all_old";
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<PutOutput<Schema>>;
+    update(
+      key: Schema & { $remove?: string[]; [string]?: any },
+      options?: {
+        conditions?: [] | {};
+        capacity?: "none" | "total" | "indexes";
+        metrics?: "none" | "size";
+        returnValues?:
+          | "none"
+          | "all_old"
+          | "updated_old"
+          | "all_new"
+          | "updated_new";
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<UpdateOutput<Schema>>;
+    query();
+    scan();
+  }
+
+  // Table
+  interface TableAttributes {
+    [attribute: string]: AnyDynamoDBType;
+  }
+  interface Indexes {
+    [index: string]: { partitionKey?: string; sortKey?: string };
+  }
+  interface TableConstructor {
+    name: string;
+    alias?: string;
+    partitionKey: string;
+    sortKey?: string;
+    entityField?: boolean | string;
+    attributes?: TableAttributes;
+    indexes?: Indexes;
+    autoExecute?: boolean;
+    autoParse?: boolean;
+    DocumentClient?: DocumentClient;
+    entities?: {}; // improve - not documented
+  }
+
+  // Entity
+  interface EntityAttributesConfiguration {
+    type?: DynamoDBType;
+    coerce?: boolean;
+    default?: string | boolean | number | [] | {} | ((data: any) => any); // Value "type" or a function
+    onUpdate?: boolean;
+    hidden?: boolean;
+    required?: boolean | "always";
+    alias?: string;
+    map?: string;
+    setType?: DynamoDBStringType | DynamoDBNumberType | DynamoDBBinaryType;
+    partitionKey?: boolean | string;
+    sortKey?: boolean | string;
```

Comment:
I think they have to be booleans, where did you see an example of string?

Reply:
here in the doc :
<img width="1123" alt="Screenshot 2020-05-04 at 08 41 15" src="https://user-images.githubusercontent.com/14938214/80941550-0b021580-8de3-11ea-8a9a-5247e755deff.png">